server/datastore/mysql/apple_mdm.go

```diff
@@ -1126,14 +1324,20 @@ SELECT
 					1 FROM host_mdm_apple_profiles hmap
 				WHERE
 					h.uuid = hmap.host_uuid
-					AND(hmap.status = 'failed'
-						OR hmap.status = 'pending')) THEN
+					AND (hmap.status IS NULL OR hmap.status != 'applied')) THEN
 			1
 		END) AS applied
 FROM
 	hosts h
 WHERE
 	%s`
+	// TODO(mna): what if there's no row in host_mdm_apple_profiles? Should it report as "applied"?
+	// If so I think the "applied" case should only be if NOT EXISTS a case with status != "applied".
+	// Not changing it until confirmed because there's a test (TestIntegrationsMDM/TestProfileManagement)
+	// that's failing with that change, so it seems it is behaving as intended as it is (i.e. the host
+	// must have at least one profile to be reported as applied/pending/failed, which makes sense too,
+	// it's just that if it had one and it got removed, once successfully removed that host would
+	// not show up anymore in those stats).
```

Comment:
If anyone can confirm that this is either working as intended, or needs to support the "host has no profiles" case as "applied" (in which case let's have it as part of another PR/ticket).

Reply:
https://github.com/fleetdm/fleet/issues/10122#issuecomment-1479590015

Reply:
Confirmed during standup that this is working as intended, will remove the comment!